### PR TITLE
DTA Followups - remove redundant methods

### PR DESCRIPTION
**Pull request by jbrockmendel**
Both arrays.datetimes and arrays.timedeltas have a `_to_m8` function.  The timedeltas one is removed since it is unnecessary, the datetimes one is given a more accurate name `_to_M8`

A couple of unused funcs from `tm` are removed.

A test is parametrized.

### CODE REVIEW

File: pandas/tests/test_base.py [hunk 1]
```diff
@@ -400,98 +400,98 @@ def test_value_counts_unique_nunique(self):
 
             assert o.nunique() == len(np.unique(o.values))
 
-    def test_value_counts_unique_nunique_null(self):
+    @pytest.mark.parametrize('null_obj', [np.nan, None])
```
Comment (jreback) [inline pandas/tests/test_base.py]: yeah this is a nasty test

File: pandas/core/arrays/datetimes.py [hunk 2]
```diff
@@ -34,7 +34,7 @@
 _midnight = time(0, 0)
 
 
-def _to_m8(key, tz=None):
+def _to_M8(key, tz=None):
```
Comment (jreback) [inline pandas/core/arrays/datetimes.py]: isn't the conversion just (line 50)

```
In [1]: from pandas._libs.tslibs import conversion

In [2]: ts = pd.Timestamp('20181001', tz='UTC')

In [3]: np.int64(conversion.pydt_to_i8(ts)).view('M8[ns]')
Out[3]: numpy.datetime64('2018-10-01T00:00:00.000000000')

In [5]: ts.to_datetime64()
Out[5]: numpy.datetime64('2018-10-01T00:00:00.000000000')
```

Comment (jreback) [reply pandas/core/arrays/datetimes.py]: this routine should really just be in conversion.pyx

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. some comments.

File: pandas/core/arrays/datetimes.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arrays/datetimes.py]: I’m hoping we can do something clever see #24576.  Not sure we are internally consistent enough to do that yet though.

**Review by jreback — APPROVED** [review]

**Pull request merged**